### PR TITLE
[Tests/Catalog] Suppress partial availability warnings

### PR DESCRIPTION
**Pull request by romoore**
`- UIFont systemFontOfSize:weight:` was not introduced until iOS 8.2. Although
most of the code was already performing runtime checks on the selector, the
compiler still emits warnings for the use of the selectors.

Closes 2779

### CODE REVIEW

File: catalog/MaterialCatalog/MDCCatalogTiles.m [hunk 1]
```diff
@@ -942,8 +942,17 @@ void MDCCatalogDrawNavigationBarTile(CGRect frame, id<MDCColorScheme> colorSchem
...
       labelStyle.alignment = NSTextAlignmentCenter;
 
+      UIFont *font;
+      if ([UIFont respondsToSelector:@selector(systemFontOfSize:weight:)]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpartial-availability"
+        font = [UIFont systemFontOfSize:11 weight:UIFontWeightMedium];
+#pragma clang diagnostic pop
+      } else {
+        font = [UIFont fontWithName:@"HelveticaNeue" size:11];
```
Comment (yarneo) [inline catalog/MaterialCatalog/MDCCatalogTiles.m]: Do we usually use "HelveticaNeue" rather than systemFontOfSize (without the weight)

**Review by yarneo — APPROVED** [review]

File: catalog/MaterialCatalog/MDCCatalogTiles.m (hunk 1 above)
Comment (romoore) [reply catalog/MaterialCatalog/MDCCatalogTiles.m]: Not sure, I stole this code from other places where we perform an if-else on the selector.

Comment (romoore) [reply catalog/MaterialCatalog/MDCCatalogTiles.m]: Ahhh... but I missed the part where we just call `[UIFont systemFontOfSize:size]` for regular font weights. :-\

Comment (romoore) [reply catalog/MaterialCatalog/MDCCatalogTiles.m]: Done

**Comment by yarneo** [comment]
LGTM

**Pull request merged**